tor2web/t2w.py

```diff
@@ -678,14 +678,17 @@ def process_request(self, req):
         self.obj.address = "http://" + self.obj.onion + self.obj.uri
 
         self.obj.headers = req.headers
-
+        self.proto = b'http' if config.transport == 'HTTP' else b'https'
+        
         # we remove the x-forwarded-for header that may contain a leaked ip
         self.obj.headers.removeHeader(b'x-forwarded-for')
 
         self.obj.headers.setRawHeaders(b'host', [self.obj.onion])
         self.obj.headers.setRawHeaders(b'connection', [b'keep-alive'])
         self.obj.headers.setRawHeaders(b'accept-encoding', [b'gzip, chunked'])
-        self.obj.headers.setRawHeaders(b'x-tor2web', [b'encrypted'])
+        self.obj.headers.setRawHeaders(b'x-tor2web', [b'1'])
+        self.obj.headers.setRawHeaders(b'X-Forwarded-Host', [config.basehost] )
```

Comment:
i would suggest here writing the request.host variable that will host the actual complete host.

this way we propagate the exact complete host for which the request is forwarded, and we can see if the node was use in TRANSLATION mode or not, and to understrand also if the node is implemented with a single TRANSLATION mapping or a multiple configuration using hosts_map.txt
1) w